source/opt/types.h

```diff
@@ -517,15 +517,15 @@ class ForwardPointer : public Type {
                                                                      \
     void GetExtraHashWords(std::vector<uint32_t>*) const override {} \
   };  // namespace analysis
-DefineParameterlessType(Void, void);
-DefineParameterlessType(Bool, bool);
-DefineParameterlessType(Sampler, sampler);
-DefineParameterlessType(Event, event);
-DefineParameterlessType(DeviceEvent, device_event);
-DefineParameterlessType(ReserveId, reserve_id);
-DefineParameterlessType(Queue, queue);
-DefineParameterlessType(PipeStorage, pipe_storage);
-DefineParameterlessType(NamedBarrier, named_barrier);
+DefineParameterlessType(Void, void)
```

Comment:
I think the better fix here is to remove the semi-colon from the macro definition.  Also the comment at the end of the macro "namespace analysis" should be removed.  It seems wrong.